xdsl/transforms/canonicalization_patterns/riscv.py

```diff
@@ -334,3 +334,37 @@ def match_and_rewrite(self, op: riscv.AddOp, rewriter: PatternRewriter) -> None:
                     ),
                 ]
             )
+
+class BitwiseAndByZero(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: riscv.AndOp, rewriter: PatternRewriter):
+        """
+        rewrite pattern to optimize bitwise and by 0
+        x & 0 = 0
+        """
+
+        # check if the first operand is 0
+        if(
+            isinstance(op.rs1, OpResult) and
+            isinstance(op.rs1.op, riscv.LiOp) and
```

Comment:
```suggestion
            isinstance(op.rs1.owner, riscv.LiOp)
```
This can work as well